### ISSUE TITLE
<Plug> mappings don't work if made after vim loads (Esp RSendMotion)

### ISSUE THREAD

**Issue by BlueDrink9** [jalvesaq/Nvim-R#470]
I've reproduced minimally in terms of vimrc.

`:nmap q <Plug>RClearConsole`, then pressing `q` doesn't seem to do anything. No errors, either, so the plug mapping exists. This works in minimal repro.

`RSendMotion` doesn't work, even in minimal repro. `\m` does.

`nmap <LocalLeader>vd <Plug>RViewDF` is the same.

The mapping clearly exists:

`:map q`

> `n  q             <Plug>RSendMotion` 

When I set `verbose=99` and do `qw` on a letter, nothing is output.

Manually mapping it to what `\m` is set to works.

**Comment by BlueDrink9** [comment]
I think the issue is in mapping them after nvim-R starts. When mapping them in the vimrc before or after calling `Plug#end()` (using `vim-plug` to manage plugins), the plug mappings do work. Mapping them in a function called after nvim-R loads, however, does not work

**Comment by BlueDrink9** [comment]
Actually, calling the function once vim loads, but before starting nvim-R, also doesn't work. Calling the function in the vimrc does work.

**Comment by jalvesaq** [comment]
The documentation (section _Custom key bindings_) says:

> After changing the maps in your vimrc, you have to restart Vim.

But actually there a minor bug in RViewDF and I'm going to fix it now.

**Comment by BlueDrink9** [comment]
But I'm not changing them in my vimrc, I'm setting them in vim! Otherwise it's impossible to set <Plug> maps for _just_ R buffers

**Comment by jalvesaq** [comment]
All maps are created with the `<buffer>` argument.

Note: I'm very busy and have almost no spare time to maintain my projects on Github. When someone opens an issue about something that is working for many years, my first guess is that there is no bug at all and that the person has some configuration problem. So, instead of trying to fix a problem that I think doesn't exist, I simply wait until the user reports what he or she has managed to fix its configuration files. Maybe the problem is caused by another plugin. Did you try to disable all other plugins?

I can't understand your issue. I know that you are trying to change mappings, but the maps are customizable since about 2013. So, could you, please, better explain the context of your problem?

**Comment by BlueDrink9** [comment]
Ok, that's understandable. If nothing has changed, I'll check my config more thoroughly, although I don't know where else to look. But yes, as mentioned, I have done a minimal reproduction.

Yes. Maps created with the `<buffer>` arg only act for the buffer they are created in. Therefore creating them in `vimrc` normally will only create them for the first buffer loaded. In my case, after loading an `R` file, I do 
`:nmap <buffer> q <Plug>RClearConsole`, then pressing q doesn't seem to do anything. No errors, either, so the plug mapping exists. This works in minimal repro.

When you say "maps are customizable", does nvim-R have a custom way to set up maps just for `R` files? Or is an autocmd the recommended way?

I hope this helps you understand my question.  I appreciate your time

**Comment by jalvesaq** [comment]
It's not a good idea to start mappings in Normal mode with regular letters because they all are already mapped to something. This is the reason why all Nvim-R's key bindings start with `<LocalLeader>`. For example,  `q` is the first of two keys to start recording what will be typed next. Please, see:

```
:h recording
```

You should not use the argument `<buffer>` when defining what sequence of keys should trigger a function:
```
 nmap q <Plug>RClearConsole
```

You can customize Nvim-R's key bindings only in the vimrc.

**Comment by BlueDrink9** [comment]
So to clarify, you don't support users remapping nvim-R's `<Plug>` commands in a buffer-local fashion, without using `<localleader>`? I assume that is a limitation of the `CreateMapping` function you use?

If this is the case, I think it would be good to create a feature-request issue for it. This wouldn't be for you to fix, but for future maintainers to consider.

Other repos allow more specific customisation, without **requiring** the use of `localleader`. For example, if I have a neoterm 'send line to repl' mapping on `<leader>ss`, the current system with nvim-R prevents me from overriding that with a <Plug> mapping for `R` buffers only. Does that use-case make sense?

> It's not a good idea to start mappings [...]

That's the purpose of customization: let the user make this choice.
In this case, I don't actually use `q`, but `<c-p>` (which I don't have use for as a default map.) `q` is what I was using for simplicity of debugging.

**Comment by jalvesaq** [comment]
I'm sorry, but I really don't have time for this issue. Please, feel free to open another issue and wait until someone can help you.
